src/lib/crypto/rsa/keys.ts

```diff
@@ -0,0 +1,109 @@
+import { webcrypto } from 'one-webcrypto'
+// import { RSA_EXCHANGE_ALG, RSA_WRITE_ALG } from '../constants.js'
+// import { RsaSize, HashAlg, KeyUse, PublicKey } from '../types.js'
+import { RSA_KEY_PAIR_ALG, DEFAULT_RSA_SIZE, DEFAULT_HASH_ALG, DEFAULT_EXPORT_KEY_FORMAT } from '../constants'
+import { RsaSize, HashAlg, PublicKey, ExportKeyFormat } from '../types'
+import utils, { normalizeBase64ToBuf } from '../utils'
+import * as aes from '../aes/index'
+import { getPublicKey, getPrivateKey } from './operations'
+// import { checkValidKeyUse } from '../errors.js'
+
+export async function makeKeypair(
+  size: RsaSize = DEFAULT_RSA_SIZE,
+  hashAlg: HashAlg = DEFAULT_HASH_ALG,
+  // use: KeyUse
+): Promise<CryptoKeyPair> {
+  // checkValidKeyUse(use)
+  // const alg = use === KeyUse.Exchange ? RSA_EXCHANGE_ALG : RSA_WRITE_ALG
+  // const uses: KeyUsage[] = use === KeyUse.Exchange ? ['encrypt', 'decrypt'] : ['sign', 'verify']
+  
+  const alg = RSA_KEY_PAIR_ALG
+  const uses: KeyUsage[] = ['decrypt', 'encrypt']
+  return webcrypto.subtle.generateKey(
+    {
+      name: alg,
+      modulusLength: size,
+      publicExponent: utils.publicExponent(),
+      hash: { name: hashAlg }
+    },
+    true,
+    uses
+  )
+}
+
+function stripPrivateKeyHeader(base64Key: string): string{
+  return base64Key
+    .replace('-----BEGIN PRIVATE KEY-----\n', '')
+    .replace('\n-----END PRIVATE KEY-----', '')
+}
+
+export async function importEncryptedKeyPair(
+  publicKey: string,
+  encPrivateKey: string,
+  passphrase: string,
+  salt?: Uint8Array,
+  hashAlg: HashAlg = DEFAULT_HASH_ALG
+): Promise<CryptoKeyPair> {
+  const importedPublicKey = await importPublicKey(publicKey)
+  const passphraseKey = await aes.deriveKey(passphrase, salt)
+  const privateKey = await aes.decrypt(encPrivateKey, passphraseKey)
+  const buf = utils.base64ToArrBuf(stripPrivateKeyHeader(privateKey))
```

Comment:
Huh is this stripping actually necessary? I'd expect if it exports correctly with this header it would handle importing correctly with this header.

Reply:
Im 90% sure I couldn't get webCrytpo not to generate DOM exceptions without it, but I can try again.

Reply:
nope you were right, I removed both calls to strip
Notably this won't work with PEM strings, it'd be up to a user to make sure they remove those headers if they want to use PEM strings.